src/nccl_ofi_rdma.c

```diff
@@ -5958,6 +5940,12 @@ int nccl_net_ofi_rdma_init(const char *provider_filter,
 	}
 	eager_max_size = (size_t) ofi_nccl_eager_max_size();
 
+	if (provider_list->domain_attr->mr_key_size != 4) {
```

Comment:
I think `mr_key_size <= sizeof(((struct nccl_net_ofi_rdma_ctrl_msg *)0)-> buff_mr_key[0])` better represents what you want to enforce here.